sdk/core/Azure.Core/tests/RetriableStreamTests.cs

```diff
@@ -429,8 +434,6 @@ public MockReadStream(long length, long throwAfter = int.MaxValue, byte offset =
 
             public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
             {
-                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Stream.cs doesn't actually throw when cancellation is requested.

Reply:
This should return Task in "cancelled" state per stream docs.

Reply:
Yeah, you're right - I still needed to remove the check here because I need to simulate cancellation being requested after the read is already in progress. In stream.cs, the token is checked before reading, but then not checked after that -https://github.com/microsoft/referencesource/blob/master/mscorlib/system/io/stream.cs#L423